File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_39916.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request
import yaml
import json
from app_runtime import redis_client


async def BenchmarkTest39916(request: Request):
    redis_value = redis_client.get('user_data')
    data = '%s' % (redis_value,)
    yaml.safe_load(data)
    return {"updated": True}

File: Benchmarks/enterprise/java/jakarta/testcode/BenchmarkTest39916.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import jakarta.ws.rs.*;
import jakarta.ws.rs.core.Context;
import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;

@Path("/")
public class BenchmarkTest39916 {

    @GET
    @Path("/BenchmarkTest39916")
    @Produces(MediaType.APPLICATION_JSON)
    public Response BenchmarkTest39916(@CookieParam("session_token") String sessionToken, @Context HttpServletRequest request, @Context HttpServletResponse response) throws Exception {
        String cookieValue = sessionToken != null ? sessionToken : "";
        StringBuilder payload = new StringBuilder();
        payload.append(cookieValue);
        String data = payload.toString();
        if (!data.isEmpty()) throw new IllegalArgumentException("invalid input: " + data);
        return Response.ok("{\"ready\":true}", MediaType.APPLICATION_JSON).build();
    }
}
